sopel/cli/run.py

```diff
@@ -345,10 +345,12 @@ def get_configuration(options):
         settings = utils.load_settings(options)
     except config.ConfigurationNotFound as error:
         print(
-            "Welcome to Sopel!\n"
-            "I can't seem to find the configuration file, "
-            "so let's generate it!\n")
-        settings = utils.wizard(error.filename)
+            "Welcome to Sopel!",
+            error,
+            "To start the configuration wizard, run `sopel configure`",
+            sep="\n",
+        )
+        sys.exit(2)
```

Comment:
That will be a hard no for me: `sys.exit` shouldn't appear here, and left to the caller to decide when to exit.

Reply:
Problem is the caller needs the exception re-raised to know that it should exit, and re-raising the exception will print the error text again in both places where this func is called—exactly the redundancy I was trying to have mal eliminate.

https://github.com/sopel-irc/sopel/blob/a174d1838e295c8c0e0749629f944c69807454bf/sopel/cli/run.py#L410-L414

https://github.com/sopel-irc/sopel/blob/a174d1838e295c8c0e0749629f944c69807454bf/sopel/cli/run.py#L608-L612

Reply:
Yes? There is nothing that prevent that code to raise a new exception with a different error message, it looks like it'll work properly and do exactly what is expected.

Reply:
Thoughts? Not redundant, no `exit()`, doesn't rely on context in a way that rubs me weird but exi says is fine.
```
$ sopel -c foobar
Welcome to Sopel! To start the configuration wizard, run `sopel configure`
Unable to find the configuration file /home/user/.sopel/foobar
```

Reply:
Outputting the solution before stating the problem is a bit weird, but tolerable. There's no rule saying this can't be changed again later, after all.

Reply:
I still don't get why raising a fresh new exception is bad, but I guess it can be changed again later, after all.

Reply:
See latest - Add `ConfNotFound.hint` for preferred order?